.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: ci
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      with:
+        fetch-depth: 0
+
+    - name: ci/setup-go
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: "1.19"
+        cache: true
+
+    - name: ci/verify-gomod
+      run: make verify-gomod
+
+    - name: ci/check-style
+      run: make check-style
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: ci/checkout-repo
+      uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+
+    - name: ci/setup-go
+      uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+      with:
+        go-version: "1.19"
+        cache: true
+
+    - name: ci/test
+      run: |
+        export PATH=$(go env GOPATH)/bin:$PATH
```

Comment:
I suggest you use the recommended way updating the GITHUB_PATH unless this is tested and works 
https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path
> You need to do this in a step before since it is not updated in place (in the same shell). It needs to be persisted so I think this command does not have any impact here

Reply:
I'd like to know why do we even need this command. `make unittest` seems like it's just doing a `go test`. Unless we are invoking some extra binaries, we don't need this.

Reply:
Honestly, I don't remember now. I'm going to remove the line and check that everything works as expected.